File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
n = int(input())

count = 0
for _ in range(n):
    s = input()
    last_index = 0
    for c in s:
        if last_index > s.index(c):
            count-=1
            break
        else:
            last_index = s.index(c)
    count+=1
print(count)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var n int
	fmt.Scanln(&n)

	var s string
	count := 0
	for i := 0; i < n; i++ {
		fmt.Scanln(&s)
		lastIndex := 0
		for _, c := range s {
			if lastIndex > strings.IndexRune(s, c) {
				count -= 1
				break
			} else {
				lastIndex = strings.IndexRune(s, c)
			}

		}
		count += 1
	}
	fmt.Print(count)
}
